### PR TITLE
fix chained findings monitor logic in update detector flow

### PR DESCRIPTION
**Pull request by eirsep**
### Description
Currently when updating a detector with aggegation rules, the chained finding doc levelmonitor is getting deleted
This PR fixes the logic to update/retain the chained  finding doc level monitor
Added to integ test to verify detector behaviour after updating detector.

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 1]
```diff
@@ -465,6 +465,18 @@ public void onResponse(Map<String, Map<String, String>> ruleFieldMappings) {
                                         new ActionListener<>() {
                                             @Override
                                             public void onResponse(Collection<IndexMonitorRequest> indexMonitorRequests) {
+                                                if (detector.getRuleIdMonitorIdMap().containsKey("chained_findings_monitor")) {
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:468]: `chained_findings_monitor` should be a constant

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 2]
```diff
@@ -465,6 +465,18 @@ public void onResponse(Map<String, Map<String, String>> ruleFieldMappings) {
...
+                                                    String cmfId = detector.getRuleIdMonitorIdMap().get("chained_findings_monitor");
+                                                    if (enabledWorkflowUsage && !indexMonitorRequests.isEmpty() && rulesById.stream().anyMatch(it -> it.getRight().isAggregationRule())) {
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:470]: Triplicate condition + same logic on line 475. Should be refactored to a private method, something like `workflowContainsAggregationRule()`

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 3]
```diff
@@ -465,6 +465,18 @@ public void onResponse(Map<String, Map<String, String>> ruleFieldMappings) {
...
+                                                        //todo update not create chained findings monitor
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:471]: Should create and link an issue for this. Or at least elaborate on why this todo exists

**Review by engechas — DISMISSED** [review]

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java (hunk 3 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:471]: Do we know why the chained monitor gets deleted in the first place?

**Review by jowg-amazon — DISMISSED** [review]

**Review by praveensameneni — DISMISSED** [review]

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java (hunk 3 above)
Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:471]: removing TODO. its resolved

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java (hunk 2 above)
Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:470]: ack

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java (hunk 1 above)
Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:468]: ack

**Review by praveensameneni — APPROVED** [review]

**Review by jowg-amazon — APPROVED** [review]

**Review by engechas — APPROVED** [review]

**Pull request merged**